File: app/src/main/java/com/nirbhor/app/ui/screens/family/CaregiverNotifyScreen.kt
package com.nirbhor.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Mail
import androidx.compose.material.icons.filled.NotificationsActive
import androidx.compose.material.icons.filled.Smartphone
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.nirbhor.app.domain.Caregiver
import com.nirbhor.app.domain.CaregiverChannel
import com.nirbhor.app.domain.DigestFrequency
import com.nirbhor.app.navigation.LocalAppContainer
import com.nirbhor.app.navigation.NavActions
import com.nirbhor.app.ui.components.NbSwitch
import com.nirbhor.app.ui.components.NirbhorTopBar
import com.nirbhor.app.ui.components.SectionLabel
import com.nirbhor.app.ui.components.TintPanel
import com.nirbhor.app.ui.i18n.tr
import com.nirbhor.app.ui.theme.Archivo
import com.nirbhor.app.ui.theme.Dimens
import com.nirbhor.app.ui.theme.NirbhorTheme
import kotlinx.coroutines.launch

/** Caregiver notification setup (4a): channels (multi), frequency, immediate-escalation override. */
@Composable
fun CaregiverNotifyScreen(actions: NavActions) {
    val colors = NirbhorTheme.colors
    val repo = LocalAppContainer.current.repository
    val scope = rememberCoroutineScope()
    val cg by repo.primaryCaregiver.collectAsStateWithLifecycle(initialValue = null)
    val caregiver = cg ?: return

    fun save(updated: Caregiver) = scope.launch { repo.upsertCaregiver(updated) }
    fun toggleChannel(ch: CaregiverChannel) {
        val next = if (ch in caregiver.channels) caregiver.channels - ch else caregiver.channels + ch
        save(caregiver.copy(channels = next))
    }

    Column(Modifier.fillMaxSize().background(colors.paper)) {
        NirbhorTopBar(title = tr("রুমানাকে জানানো", "Telling Rumana"), onBack = actions::back)
        Column(
            Modifier.fillMaxSize().verticalScroll(rememberScrollState()).padding(horizontal = Dimens.screenPadding, vertical = 16.dp),
            verticalArrangement = Arrangement.spacedBy(18.dp),
        ) {
            Column(verticalArrangement = Arrangement.spacedBy(Dimens.cardGap)) {
                SectionLabel(tr("কীভাবে জানাব", "How to tell them"))
                ChannelCard(Icons.Filled.Mail, tr("ইমেইল", "Email"), tr("বিনামূল্যে · বিস্তারিত হিসাব পাঠানো যায়", "Free · can send a full record"),
                    CaregiverChannel.EMAIL in caregiver.channels, ::toggleChannel, CaregiverChannel.EMAIL, verifiedEmail = caregiver.email.takeIf { caregiver.emailVerified })
                ChannelCard(Icons.Filled.Smartphone, tr("এসএমএস", "SMS"), tr("ইন্টারনেট ছাড়াও পৌঁছায় · ছোট বার্তা", "Reaches without internet · short message"),
                    CaregiverChannel.SMS in caregiver.channels, ::toggleChannel, CaregiverChannel.SMS)
                ChannelCard(Icons.Filled.NotificationsActive, tr("অ্যাপ নোটিফিকেশন", "App notification"), tr("তাদের ফোনে জানানো হবে", "On their phone"),
                    CaregiverChannel.APP in caregiver.channels, ::toggleChannel, CaregiverChannel.APP)
            }

            Column(verticalArrangement = Arrangement.spacedBy(Dimens.cardGap)) {
                SectionLabel(tr("কত ঘন ঘন", "How often"))
                RadioCard(tr("দিনে একবার, সব একসাথে", "Once a day, all together"), tr("রাত ৯:৩০-এ দিনের হিসাব", "Day's summary at 9:30 PM"),
                    caregiver.digestFrequency == DigestFrequency.DAILY_DIGEST) { save(caregiver.copy(digestFrequency = DigestFrequency.DAILY_DIGEST)) }
                RadioCard(tr("প্রতিবার সাথে সাথে", "Every time, right away"), tr("বেশি ইমেইল যাবে", "More emails"),
                    caregiver.digestFrequency == DigestFrequency.IMMEDIATE) { save(caregiver.copy(digestFrequency = DigestFrequency.IMMEDIATE)) }
            }

            Row(
                Modifier.fillMaxWidth().clip(RoundedCornerShape(12.dp)).background(colors.warmSoft).padding(14.dp),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Text(tr("পরপর দুবার বাদ পড়লে সাথে সাথেই জানাব", "If missed twice in a row, tell them right away"),
                    style = NirbhorTheme.type.cardTitleSecondary, color = colors.warmD, modifier = Modifier.weight(1f))
                Spacer(Modifier.width(12.dp))
                NbSwitch(checked = caregiver.escalateOnSecondMiss, onCheckedChange = { save(caregiver.copy(escalateOnSecondMiss = it)) }, onColor = colors.warmD)
            }

            TintPanel(background = colors.sage) {
                Text(tr("আপনার পরিবার যা যা বার্তা পায়, আপনি সবই দেখতে পান। সর্বশেষ পাঠানো হয়েছে গতকাল।",
                    "You can see every message your family gets. Last sent yesterday."), style = NirbhorTheme.type.body, color = colors.ink2)
            }
        }
    }
}

@Composable
private fun ChannelCard(icon: ImageVector, title: String, subtitle: String, selected: Boolean, onToggle: (CaregiverChannel) -> Unit, channel: CaregiverChannel, verifiedEmail: String? = null) {
    val colors = NirbhorTheme.colors
    val shape = RoundedCornerShape(14.dp)
    Column(
        Modifier.fillMaxWidth().clip(shape).background(colors.card)
            .border(if (selected) 2.dp else 1.5.dp, if (selected) colors.calm else colors.line, shape)
            .clickable { onToggle(channel) }.padding(16.dp),
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Box(Modifier.size(46.dp).clip(RoundedCornerShape(14.dp)).background(colors.sage), contentAlignment = Alignment.Center) {
                Icon(icon, contentDescription = null, tint = colors.ink2, modifier = Modifier.size(24.dp))
            }
            Spacer(Modifier.width(12.dp))
            Column(Modifier.weight(1f)) {
                Text(title, style = NirbhorTheme.type.cardTitlePrimary, color = colors.ink)
                Text(subtitle, style = NirbhorTheme.type.meta, color = colors.ink2)
            }
            SelIndicator(selected)
        }
        if (verifiedEmail != null) {
            Spacer(Modifier.height(10.dp))
            Row(
                Modifier.fillMaxWidth().clip(RoundedCornerShape(11.dp)).background(colors.paper).padding(horizontal = 12.dp, vertical = 10.dp),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Text(verifiedEmail, style = NirbhorTheme.type.meta.copy(fontFamily = Archivo), color = colors.ink, overflow = TextOverflow.Ellipsis, modifier = Modifier.weight(1f))
                Spacer(Modifier.width(8.dp))
                Icon(Icons.Filled.Check, contentDescription = null, tint = colors.calmD, modifier = Modifier.size(16.dp))
                Text(tr("যাচাই হয়েছে", "Verified"), style = NirbhorTheme.type.statusPill, color = colors.calmD)
            }
        }
    }
}

@Composable
private fun SelIndicator(selected: Boolean) {
    val colors = NirbhorTheme.colors
    Box(
        Modifier.size(28.dp).clip(CircleShape)
            .background(if (selected) colors.calm else androidx.compose.ui.graphics.Color.Transparent)
            .border(if (selected) 0.dp else 2.dp, if (selected) androidx.compose.ui.graphics.Color.Transparent else colors.line, CircleShape),
        contentAlignment = Alignment.Center,
    ) {
        if (selected) Icon(Icons.Filled.Check, contentDescription = null, tint = colors.paper, modifier = Modifier.size(18.dp))
    }
}

@Composable
private fun RadioCard(title: String, subtitle: String, selected: Boolean, onClick: () -> Unit) {
    val colors = NirbhorTheme.colors
    val shape = RoundedCornerShape(14.dp)
    Row(
        Modifier.fillMaxWidth().clip(shape).background(colors.card)
            .border(if (selected) 2.dp else 1.5.dp, if (selected) colors.calm else colors.line, shape)
            .clickable(onClick = onClick).padding(16.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Column(Modifier.weight(1f)) {
            Text(title, style = NirbhorTheme.type.cardTitleSecondary, color = colors.ink)
            Text(subtitle, style = NirbhorTheme.type.meta, color = colors.ink3)
        }
        Box(Modifier.size(21.dp).clip(CircleShape).border(if (selected) 6.dp else 2.dp, if (selected) colors.calm else colors.line, CircleShape))
    }
}

File: lib/ui/screens/caregiver_notify_screen.dart
import 'package:flutter/material.dart';

import '../../data/app_scope.dart';
import '../../domain/models.dart';
import '../../navigation/nav_actions.dart';
import '../../theme/theme.dart';
import '../components/controls.dart';
import '../components/labels.dart';
import '../components/scaffold.dart';
import '../components/surfaces.dart';

/// Caregiver notification settings (4a) — which channels are on, how often, and the escalation.
class CaregiverNotifyScreen extends StatelessWidget {
  const CaregiverNotifyScreen({super.key});

  @override
  Widget build(BuildContext context) {
    final colors = context.colors;
    return Scaffold(
      backgroundColor: colors.paper,
      body: RepoBuilder<Caregiver?>(
        query: (repo) => repo.primaryCaregiver(),
        loading: Column(
          children: [
            NirbhorTopBar(
              title: context.tr('কাকে জানানো হবে', 'Caregiver notifications'),
              onBack: context.nav.back,
            ),
          ],
        ),
        builder: (context, caregiver) {
          if (caregiver == null) {
            return Column(
              children: [
                NirbhorTopBar(
                  title: context.tr('কাকে জানানো হবে', 'Caregiver notifications'),
                  onBack: context.nav.back,
                ),
                Padding(
                  padding: const EdgeInsets.all(Dimens.screenPadding),
                  child: TintPanel(
                    background: colors.sage,
                    child: Text(
                      context.tr('আগে একজন যত্নকারী যুক্ত করুন।', 'Add a caregiver first.'),
                      style: context.type.body.copyWith(color: colors.ink2),
                    ),
                  ),
                ),
              ],
            );
          }

          Future<void> save(Caregiver updated) => context.repo.upsertCaregiver(updated);
          void toggleChannel(CaregiverChannel ch) {
            final next = Set<CaregiverChannel>.of(caregiver.channels);
            if (!next.remove(ch)) next.add(ch);
            save(caregiver.copyWith(channels: next));
          }

          return Column(
            children: [
              NirbhorTopBar(
                title: context.tr(
                  '${caregiver.name}-কে জানানো',
                  'Telling ${caregiver.name}', hi: '${caregiver.name} को बताना', es: 'Avisar a ${caregiver.name}',
                ),
                onBack: context.nav.back,
              ),
              Expanded(
                child: SingleChildScrollView(
                  padding: const EdgeInsets.symmetric(
                    horizontal: Dimens.screenPadding,
                    vertical: 16,
                  ),
                  child: SafeArea(
                    top: false,
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.stretch,
                      children: [
                        SectionLabel(context.tr('কীভাবে জানাব', 'How to tell them')),
                        const SizedBox(height: Dimens.cardGap),
                        _ChannelCard(
                          icon: Icons.mail,
                          title: context.tr('ইমেইল', 'Email'),
                          subtitle: context.tr(
                            'বিনামূল্যে · বিস্তারিত হিসাব পাঠানো যায়',
                            'Free · can send a full record',
                          ),
                          selected: caregiver.channels.contains(CaregiverChannel.email),
                          onTap: () => toggleChannel(CaregiverChannel.email),
                          verifiedEmail: caregiver.emailVerified ? caregiver.email : null,
                        ),
                        const SizedBox(height: Dimens.cardGap),
                        _ChannelCard(
                          icon: Icons.smartphone,
                          title: context.tr('এসএমএস', 'SMS'),
                          subtitle: context.tr(
                            'ইন্টারনেট ছাড়াও পৌঁছায় · ছোট বার্তা',
                            'Reaches without internet · short message',
                          ),
                          selected: caregiver.channels.contains(CaregiverChannel.sms),
                          onTap: () => toggleChannel(CaregiverChannel.sms),
                        ),
                        const SizedBox(height: Dimens.cardGap),
                        _ChannelCard(
                          icon: Icons.notifications_active,
                          title: context.tr('অ্যাপ নোটিফিকেশন', 'App notification'),
                          subtitle: context.tr('তাদের ফোনে জানানো হবে', 'On their phone'),
                          selected: caregiver.channels.contains(CaregiverChannel.app),
                          onTap: () => toggleChannel(CaregiverChannel.app),
                        ),
                        const SizedBox(height: 18),

                        SectionLabel(context.tr('কত ঘন ঘন', 'How often')),
                        const SizedBox(height: Dimens.cardGap),
                        _RadioCard(
                          title: context.tr(
                            'দিনে একবার, সব একসাথে',
                            'Once a day, all together',
                          ),
                          subtitle: context.tr(
                            'রাত ৯:৩০-এ দিনের হিসাব',
                            "Day's summary at 9:30 PM",
                          ),
                          selected: caregiver.digestFrequency == DigestFrequency.dailyDigest,
                          onTap: () => save(
                            caregiver.copyWith(digestFrequency: DigestFrequency.dailyDigest),
                          ),
                        ),
                        const SizedBox(height: Dimens.cardGap),
                        _RadioCard(
                          title: context.tr('প্রতিবার সাথে সাথে', 'Every time, right away'),
                          subtitle: context.tr('বেশি ইমেইল যাবে', 'More emails'),
                          selected: caregiver.digestFrequency == DigestFrequency.immediate,
                          onTap: () => save(
                            caregiver.copyWith(digestFrequency: DigestFrequency.immediate),
                          ),
                        ),
                        const SizedBox(height: 18),

                        Container(
                          decoration: BoxDecoration(
                            color: colors.warmSoft,
                            borderRadius: BorderRadius.circular(12),
                          ),
                          padding: const EdgeInsets.all(14),
                          child: Row(
                            children: [
                              Expanded(
                                child: Text(
                                  context.tr(
                                    'পরপর দুবার বাদ পড়লে সাথে সাথেই জানাব',
                                    'If missed twice in a row, tell them right away',
                                  ),
                                  style: context.type.cardTitleSecondary
                                      .copyWith(color: colors.warmD),
                                ),
                              ),
                              const SizedBox(width: 12),
                              NbSwitch(
                                checked: caregiver.escalateOnSecondMiss,
                                onColor: colors.warmD,
                                onChanged: (v) =>
                                    save(caregiver.copyWith(escalateOnSecondMiss: v)),
                              ),
                            ],
                          ),
                        ),
                        const SizedBox(height: 18),

                        TintPanel(
                          background: colors.sage,
                          child: Text(
                            context.tr(
                              'আপনার পরিবার যা যা বার্তা পায়, আপনি সবই দেখতে পান। সর্বশেষ পাঠানো হয়েছে গতকাল।',
                              'You can see every message your family gets. Last sent yesterday.',
                            ),
                            style: context.type.body.copyWith(color: colors.ink2),
                          ),
                        ),
                      ],
                    ),
                  ),
                ),
              ),
            ],
          );
        },
      ),
    );
  }
}

class _ChannelCard extends StatelessWidget {
  const _ChannelCard({
    required this.icon,
    required this.title,
    required this.subtitle,
    required this.selected,
    required this.onTap,
    this.verifiedEmail,
  });

  final IconData icon;
  final String title;
  final String subtitle;
  final bool selected;
  final VoidCallback onTap;
  final String? verifiedEmail;

  @override
  Widget build(BuildContext context) {
    final colors = context.colors;
    return Semantics(
      checked: selected,
      child: GestureDetector(
        behavior: HitTestBehavior.opaque,
        onTap: onTap,
        child: Container(
          decoration: BoxDecoration(
            color: colors.card,
            borderRadius: BorderRadius.circular(14),
            border: Border.all(
              color: selected ? colors.calm : colors.line,
              width: selected ? 2 : 1.5,
            ),
          ),
          padding: const EdgeInsets.all(16),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.stretch,
            children: [
              Row(
                children: [
                  Container(
                    width: 46,
                    height: 46,
                    alignment: Alignment.center,
                    decoration: BoxDecoration(
                      color: colors.sage,
                      borderRadius: BorderRadius.circular(14),
                    ),
                    child: Icon(icon, size: 24, color: colors.ink2),
                  ),
                  const SizedBox(width: 12),
                  Expanded(
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Text(
                          title,
                          style: context.type.cardTitlePrimary.copyWith(color: colors.ink),
                        ),
                        Text(
                          subtitle,
                          style: context.type.meta.copyWith(color: colors.ink2),
                        ),
                      ],
                    ),
                  ),
                  CheckCircle(selected: selected, size: 28),
                ],
              ),
              if (verifiedEmail != null) ...[
                const SizedBox(height: 10),
                Container(
                  decoration: BoxDecoration(
                    color: colors.paper,
                    borderRadius: BorderRadius.circular(11),
                  ),
                  padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 10),
                  child: Row(
                    children: [
                      Expanded(
                        child: Text(
                          verifiedEmail!,
                          // An email address is a Latin run whatever the active locale.
                          style: context.type
                              .asLatin(context.type.meta)
                              .copyWith(color: colors.ink),
                          overflow: TextOverflow.ellipsis,
                        ),
                      ),
                      const SizedBox(width: 8),
                      Icon(Icons.check, size: 16, color: colors.calmD),
                      Text(
                        context.tr('যাচাই হয়েছে', 'Verified'),
                        style: context.type.statusPill.copyWith(color: colors.calmD),
                      ),
                    ],
                  ),
                ),
              ],
            ],
          ),
        ),
      ),
    );
  }
}

class _RadioCard extends StatelessWidget {
  const _RadioCard({
    required this.title,
    required this.subtitle,
    required this.selected,
    required this.onTap,
  });

  final String title;
  final String subtitle;
  final bool selected;
  final VoidCallback onTap;

  @override
  Widget build(BuildContext context) {
    final colors = context.colors;
    return Semantics(
      selected: selected,
      inMutuallyExclusiveGroup: true,
      child: GestureDetector(
        behavior: HitTestBehavior.opaque,
        onTap: onTap,
        child: Container(
          decoration: BoxDecoration(
            color: colors.card,
            borderRadius: BorderRadius.circular(14),
            border: Border.all(
              color: selected ? colors.calm : colors.line,
              width: selected ? 2 : 1.5,
            ),
          ),
          padding: const EdgeInsets.all(16),
          child: Row(
            children: [
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      title,
                      style: context.type.cardTitleSecondary.copyWith(color: colors.ink),
                    ),
                    Text(subtitle, style: context.type.meta.copyWith(color: colors.ink3)),
                  ],
                ),
              ),
              Container(
                width: 21,
                height: 21,
                decoration: BoxDecoration(
                  shape: BoxShape.circle,
                  border: Border.all(
                    color: selected ? colors.calm : colors.line,
                    width: selected ? 6 : 2,
                  ),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
